markets/retrievaladapter/blockstoremanager.go

```diff
@@ -0,0 +1,199 @@
+package retrievaladapter
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/xerrors"
+
+	"github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-cid"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-merkledag"
+	unixfile "github.com/ipfs/go-unixfs/file"
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/blockstore"
+
+	retrievalimpl "github.com/filecoin-project/go-fil-markets/retrievalmarket/impl"
+)
+
+// BlockstoreManager provides an interface to get a blockstore and perform
+// transformations based on a DAG root CID.
+// It is used to provide the retrieval client with a blockstore backed by
+// either an IPFS or CARv2 file.
+type BlockstoreManager interface {
+	retrievalimpl.BlockstoreAccessor
+	WriteCarV1(ctx context.Context, rootCid cid.Cid, out string) error
+	WriteUnixFSFile(ctx context.Context, rootCid cid.Cid, out string) error
```

Comment:
This seems odd. Why do these methods _need_ to be here? This logic for writing a CARv1 or a UnixFS file will not change across implementations AFAIK.

The input for those methods is a root CID and a **blockstore**. The caller can get the blockstore from the `BlockstoreAccessor`, and it has the root CID in its local scope.

IMO these should be top-level utility functions taking a blockstore, not members of an interface.